### DISCUSSION TITLE
KeyCloak docker-compose

### DISCUSSION THREAD

**Discussion by bluezebra-developers** [apache/camel-karavan#1458] [Q&A]
### Ask a question

Hello,

Could you please advise how to properly configure Keycloak in a single network?
When deploying the container and the code reaches the /ui/users/me endpoint, I receive a 500 error.

Browser
Main AuthType oidc
SsoApi User is now authenticated.
GET http://localhost:8080/ui/users/me 500 (Internal Server Error)

karavan
2024-11-22 WARN  [io.qua.oid.run.OidcRecorder] (vert.x-eventloop-thread-3) OIDC server is not available at the 'http://localhost:8079/realms/karavan' URL. Please make sure it is correct. Note it has to end with a realm value if you work with Keycloak, for example: 'https://localhost:8180/auth/realms/quarkus'
2024-11-22 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (vert.x-eventloop-thread-3) HTTP Request to /ui/users/me failed, error id: ab6d0644-18bb-4fe1-98e9-0f33808d8a60-2: io.quarkus.oidc.OIDCException: OIDC Server is not available

`services:
  karavan:
    container_name: karavan
    image: ghcr.io/apache/camel-karavan:4.8.0-oidc
    ports:
      - "8080:8080"
    environment:
      - KARAVAN_GIT_REPOSITORY=${KARAVAN_GIT_REPOSITORY}
      - KARAVAN_GIT_USERNAME=${KARAVAN_GIT_USERNAME}
      - KARAVAN_GIT_PASSWORD=${KARAVAN_GIT_PASSWORD}
      - KARAVAN_GIT_BRANCH=main
      - KARAVAN_CONTAINER_IMAGE_REGISTRY=registry:5000
      - KARAVAN_CONTAINER_IMAGE_REGISTRY_USERNAME=root
      - KARAVAN_CONTAINER_IMAGE_REGISTRY_PASSWORD=root
      - KARAVAN_CONTAINER_IMAGE_GROUP=karavan
      - KARAVAN_DOCKER_NETWORK=karavan
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock"
    labels:
      - "org.apache.camel.karavan/type=internal"
    networks:
      - karavan

  registry:
    container_name: registry
    image: registry:2
    restart: always
    ports:
      - "5555:5000"
    labels:
      - "org.apache.camel.karavan/type=internal" 
    networks:
      - karavan

  keycloak:
    container_name: keycloak
    image: quay.io/keycloak/keycloak:latest
    ports:
      - "8079:8080"
    environment:
      - KC_BOOTSTRAP_ADMIN_USERNAME=admin
      - KC_BOOTSTRAP_ADMIN_PASSWORD=admin
    command:
      - start-dev
    networks:
      - karavan

networks:
  karavan:
    name: karavan
`

### Variant

None

### Container Management (if applicable)

Docker

### Operating System (if applicable)

Linux

### Version

4.8.0

**Comment by mgubaidullin** [comment]
You need to configure connection to keycloak using the following environment variables
```
KARAVAN_KEYCLOAK_URL=http://$YOUR_KEYCLOAK_HOST:$PORT
KARAVAN_KEYCLOAK_REALM=karavan
KARAVAN_KEYCLOAK_FRONTEND_CLIENTID=$FRONTEND_CLIENTID
KARAVAN_KEYCLOAK_BACKEND_CLIENTID=$BACKEND_CLIENTID
KARAVAN_KEYCLOAK_BACKEND_SECRET=$BACKEND_CLIENT_SECRET
```
Explanation of Variables:

1. KARAVAN_KEYCLOAK_URL: Replace $YOUR_KEYCLOAK_HOST and $PORT with the appropriate values for your Keycloak instance (not localhost).
2. KARAVAN_KEYCLOAK_REALM: This should be set to karavan or the specific realm you are using for your configuration.
3. KARAVAN_KEYCLOAK_FRONTEND_CLIENTID: Replace $FRONTEND_CLIENTID with the client ID configured for the frontend in Keycloak.
4. KARAVAN_KEYCLOAK_BACKEND_CLIENTID: Replace $BACKEND_CLIENTID with the client ID configured for the backend in Keycloak.
5. KARAVAN_KEYCLOAK_BACKEND_SECRET: Replace $BACKEND_CLIENT_SECRET with the secret key associated with the backend client.
